### DISCUSSION TITLE
scanner

### DISCUSSION THREAD

**Discussion by luc-git** [OpenPrinting/cups#870] [General]
Hello there just a question.  Why cups doesn't supports scanner while ipp protocol supports it?

https://openprinting.github.io/gsoc2019/02-ipp-scan-server/

**Comment by michaelrsweet** [comment]
First, IPP Scan isn't supported in shipping printers or scanners - eSCL is the industry standard.

Second, the current CUPS architecture doesn't lend itself to supporting scanning.

Finally, there is work towards adding IPP Scan and eSCL support to PAPPL, and SANE already supports both.

**Reply by luc-git** [reply]
What are shipping printers or scanners?

**Reply by michaelrsweet** [reply]
Printers or scanners you can buy at the store.
